File: 백준/Silver/11722. 가장 긴 감소하는 부분 수열/가장 긴 감소하는 부분 수열.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] a = new int[n];
        int[] dp = new int[n];
        int res = Integer.MIN_VALUE;
        String[] str = br.readLine().split(" ");
        for (int i = 0; i < n; i++)
            a[i] = Integer.parseInt(str[i]);

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < i; j++) {
                if (a[i] < a[j])
                    dp[i] = Math.max(dp[i], dp[j]);
            }
            dp[i] += 1;
            res = Math.max(res, dp[i]);
        }

        System.out.println(res);
    }
}

File: 백준/Silver/11722. 가장 긴 감소하는 부분 수열/가장 긴 감소하는 부분 수열.py
N = int(input())
A = [*map(int, input().split())]
dp = [0] * N
for i in range(0, N):
    for j in range(0, i):
        if A[i] < A[j]:
            dp[i] = max(dp[i], dp[j])
    dp[i] += 1
print(max(dp))